crates/services/gas_price_service/src/v1/da_source_service/block_committer_costs.rs

```diff
@@ -103,7 +101,10 @@ where
                         .get_costs_by_l2_block_number(*next_height.deref())
                         .await?
                 }
-                None => self.client.get_latest_costs().await?.into_iter().collect(),
+                None => {
+                    // prevent calling the latest endpoint
+                    return Err(anyhow!("last_recorded_height is None"));
```

Comment:
Won't this always throw an error on startup until we get an L2 block and call `handle_normal_block`

Reply:
nop, see crates/services/gas_price_service/src/v1/uninitialized_task.rs, we set it before we start the service.